src/models/User.js

```diff
@@ -4,91 +4,101 @@ const jwt = require('jsonwebtoken')
 const bcrypt = require('bcryptjs')
 require('dotenv').config()
 
-const userSchema = mongoose.Schema(
-    {
-        name: {
-            type: String,
-            trim: true,
-            required: true,
-        },
-        email: {
-            type: String,
-            trim: true,
-            required: true,
-            unique: true,
-            validate(value) {
-                if (!validator.isEmail(value)) {
-                    throw new Error('Email is invalid')
-                }
-            },
-        },
-        passsword: {
-            defaultValue: false,
-            type: String,
-            trim: true,
-            required: true,
-            minlength: 8,
-            validate(value) {
-                var strength = 0
-                if (value.match(/[a-z]+/)) {
-                    strength += 1
-                }
-                if (value.match(/[A-Z]+/)) {
-                    strength += 1
-                }
-                if (value.match(/[0-9]+/)) {
-                    strength += 1
-                }
-                if (value.match(/[#$&!@]+/)) {
-                    strength += 1
-                }
-                if (strength < 4) {
-                    throw new Error(
-                        'The password must contain a mix of uppercase and lowercase alphabets along with numbers and special chacracters'
-                    )
-                }
-            },
-        },
-        phoneNumber: {
-            type: String,
-            trim: true,
-            required: true,
-            validate(value) {
-                if (!validator.isMobileNumber(value)) {
-                    throw new Error('Invalid Mobile Number')
-                }
-            },
-        },
-        tokens: [
+
+const userSchema = mongoose.Schema({
+    name:{
+        type:String,
+        trim:true,
+        required:true
+    },
+    email:{
+        type:String,
+        trim:true,
+        required:true,
+        unique:true,
+        validate(value)
+        {
+            if(!validator.isEmail(value))
+            {
+                throw new Error('Email is invalid')
+            }
+        }
+    },
+    password:{
+        type:String,
+        trim:true,
+        required:true,
+        minlength:8,
+        validate(value)
+        {
```

Comment:
Make a Utility Function for password validation and just call it here. We might have to use the same function again later.

Reply:
Yes, I have done that now